pkg/gce-pd-csi-driver/node.go

```diff
@@ -322,28 +318,17 @@ func (ns *GCENodeServer) GetVolumeLimits() (int64, error) {
 	// Default value 16 for memory lower than cutoff
 	volumeLimits := volumeLimit16
 
-	// Machine-type format: n1-type-CPUS or custom-CPUS-RAM
+	// Machine-type format: n1-type-CPUS or custom-CPUS-RAM or f1/g1-type
 	machineType := ns.MetadataService.GetMachineType()
 	if strings.HasPrefix(machineType, "n1-") {
-		// Predefined types with memory higher than cutoff
+		// Predefined types with limits 128
 		volumeLimits = volumeLimit128
 	} else if machineType == "f1-micro" || machineType == "g1-small" {
-		// Predefined types with memory lower than cutoff
+		// Predefined types with limits 16
 		return volumeLimits, nil
 	} else if strings.HasPrefix(machineType, "custom-") {
 		// Custom machine types
-		splits := strings.Split(machineType, "-")
-		if len(splits) != customTypeParts {
-			return volumeLimits, fmt.Errorf("Unexpected machine type: %s", machineType)
-		}
-		memStr := splits[customTypeMem]
-		mem, err := strconv.Atoi(memStr)
-
-		if (err == nil) && (mem >= memCutOff) {
-			volumeLimits = volumeLimit128
-		} else {
-			return volumeLimits, err
-		}
+		volumeLimits = volumeLimit128
 	} else {
 		return volumeLimits, fmt.Errorf("Unexpected machine type: %s", machineType)
```

Comment:
Maybe this can be simplified to a case statement.  Where if it's custom or n1, then it's 128.  Everything else is 16.  I think we don't need to error for unexpected machine type. That will hopefully make this more resilient to any new machine types added in the future.

Reply:
Yes, it makes sense